MIPs/new-protocol_metadata.md

```diff
@@ -0,0 +1,159 @@
+  - MIP-defined hierarchical types, such as:
+    - 'MIP3:Person:Name'
+    - 'MIP3:Address'
+    - 'MIP3:Home:MultiAddress'
+  - Application-defined hierarchical property name, such as:
+    - 'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Actor.Username'
+    - 'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Sparks'
+  - Or an Atomic type defined above
+
+### Schema
+
+A Mercury Schema is a set of properties (with types) that are either present in a given profile or not. If required properties are missing, then that profile does not match/represent this schema. There can be optional properties in a schema that might be present or not. If a property exists with the same key, but different type, then it is not a match. For optional properties, these non-matches must be treated as missing, for required properties, such a non-match means failure of matching the whole schema.
+
+By representing the schema as a flat list of properties developers can easily combine MIP based shared schemas with their own dApp defined schemas into a single server request for metadata or alternately a fine grained composition tailored to the specific needs of a view in a UI.
+
+The schema should not be transferred together with the profile, it is an independent definition. Applications and use-cases should define schemas and match them to profiles. So profiles are not strongly typed, and just looking at a profile it does not need to describe what schemas it implements.
+
+## Specification
+
+Cap'n Proto partial definition -
+
+```capnproto
+struct MetaProperty {
+    # typed key for key-value dictionary/map container
+
+    key @0 :Text;
+    # The unique name of the property in the mapping. This name can have multiple forms:
+    #   - MIP defined, ex. 'MIP3.Persona.Username', 'MIP3.Home.Addresses',
+    #        'MIP3:Home:MultiAddress'
+    #   - dApp defined, ex. 'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Actor.Username',
+    #        'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Sparks'
+
+    atomic @1 :Text;
+    # semantic is a string that represents an optional codec-like type definition
+    # semantic types have similar forms to Keys:
+    #   - MIP defined, ex. 'MIP3:Person:Name', 'MIP3:Address', 'MIP3:Home:MultiAddress'
+    #   - dApp defined, ex. 'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Actor.Username',
+    #       'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Sparks'
+    #   - An Atomic type defined above
+
+    semantic @1 :Text;
+    # semantic is a string that represents an optional codec-like type definition
+    # semantic types have similar forms to Keys:
+    #   - MIP defined, ex. 'MIP3:Person:Name', 'MIP3:Address', 'MIP3:Home:MultiAddress'
+    #   - dApp defined, ex. 'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Actor.Username',
+    #       'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Sparks'
+    #   - An Atomic type defined above
```

Comment:
I think there's some copy-paste duplication confusion in the comments of `atomic` vs `semantic`, but I think I still get the point despite of that.